tinymce-angular-component/src/main/ts/utils/ScriptLoader.ts

```diff
@@ -15,41 +15,66 @@ export interface IStateObj {
   scriptLoaded: boolean;
 }
 
-const injectScriptTag = (scriptId: string, doc: Document, url: string, callback: callbackFn) => {
-  const scriptTag = doc.createElement('script');
-  scriptTag.referrerPolicy = 'origin';
-  scriptTag.type = 'application/javascript';
-  scriptTag.id = scriptId;
-  scriptTag.addEventListener('load', callback);
-  scriptTag.src = url;
-  if (doc.head) {
-    doc.head.appendChild(scriptTag);
-  }
-};
-
-const create = (): IStateObj => {
+const createState = (): IStateObj => {
   return {
     listeners: [],
     scriptId: uuid('tiny-script'),
     scriptLoaded: false
   };
 };
 
-const load = (state: IStateObj, doc: Document, url: string, callback: callbackFn) => {
-  if (state.scriptLoaded) {
-    callback();
-  } else {
-    state.listeners.push(callback);
-    if (!doc.getElementById(state.scriptId)) {
-      injectScriptTag(state.scriptId, doc, url, () => {
-        state.listeners.forEach((fn) => fn());
-        state.scriptLoaded = true;
-      });
+interface ScriptLoader {
+  load: (doc: Document, url: string, callback: callbackFn) => void;
+  reinitialize: () => void;
+}
+
+const CreateScriptLoader = (): ScriptLoader => {
+  let state: IStateObj = createState();
+
+  const injectScriptTag = (scriptId: string, doc: Document, url: string, callback: callbackFn) => {
+    const scriptTag = doc.createElement('script');
+    scriptTag.referrerPolicy = 'origin';
+    scriptTag.type = 'application/javascript';
+    scriptTag.id = scriptId;
+    scriptTag.src = url;
+
+    const handler = () => {
+      callback();
+      scriptTag.removeEventListener('load', handler);
```

Comment:
I'd suggest putting the remove event listener first, as that way it ensures it'll be removed, as if the callback throws an exception then this won't ever be called.